File: 2357. Make Array Zero by Subtracting Equal Amounts/2357.java
// 2357. Make Array Zero by Subtracting Equal Amounts solution in Java
// https://leetcode.com/problems/make-array-zero-by-subtracting-equal-amounts/description/

class Solution {
    public int minimumOperations(int[] nums) {
        Set<Integer> nums_set = new HashSet<>();

        for (int num : nums) {
            if (num > 0) nums_set.add(num);
        }

        return nums_set.size();
    }
}

File: 2357. Make Array Zero by Subtracting Equal Amounts/2357.go
// 2357. Make Array Zero by Subtracting Equal Amounts solution in GoLang
// https://leetcode.com/problems/make-array-zero-by-subtracting-equal-amounts/description/

func minimumOperations(nums []int) int {
    nums_map := make(map[int]bool)

    for _, num := range nums {
        if num != 0 {
            nums_map[num] = true
        }
    }

    return len(nums_map)
}